tests/test_default_qubit.py

```diff
@@ -483,6 +483,20 @@ def test_sample_dimensions(self):
         s3 = self.dev.sample('CZ', [0,1], [], 17)
         self.assertAllEqual(s3.shape, (17,))
 
+    def test_sample_mean_and_variance(self):
+        self.logTestName()
+        self.dev.reset()
+
+        self.dev.apply('RX', wires=[0], par=[1.570796])
+
+        s1 = self.dev.sample('PauliZ', [0], [], 2000)
+
+        # We use large deltas here because sampling is stochastic
+        # this ensures that the test only fails if something in
+        # the sampling itself goes awry
+        self.assertAlmostEqual(s1.mean(), 0.0, delta=0.1)
```

Comment:
What do you think about statistical tests like those? Theoretically they could fail even if there is nothing wrong in the code, just because the sampling is random.

Reply:
I dislike them a lot 😆 Often because it is hard to tell how often they fail; attempts to mitigate this can either cause the test to take an insignificant amount of time, or the tolerance value to appear to be a 'magic number'.

But, with testing stochastic code, it is usually unavoidable. One thing you could do is make the tolerance value a set number of standard deviations from the sample population mean, i.e., `delta=num_sd/np.sqrt(shots)`.

Reply:
That's a good idea, then I'll do this.

Reply:
Actually, the standard deviation here is 1, which makes giving the delta in terms of the standard deviation quite nonsensical. I think 0.1 should be fine for 2000 samples -- I calculated that the probability that the estimate lies outside of this value range is O(10^-5).

Reply:
:+1: